internal/appconfig/testdata/validate-duplicate-services-diff-processgroup.toml

```diff
@@ -0,0 +1,25 @@
+app = "foo"
+
+[processes]
+web = "/run1"
+app = "/run2"
+
+
+[http_service]
+  internal_port = 1738
+  processes = ["app"]
+
+[[services]]
+  internal_port = 1738
+  protocol = "tcp"
+  processes = ["web"]
```

Comment:
I would add a more complete case,  for example a service that belongs to two process groups, and then another service that only belong to one of them.  Both services listen on tcp-1738, and the validation must fail only for the process group that is common to both services.